homeassistant/components/nextdns/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Adds config flow for NextDNS."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from aiohttp.client_exceptions import ClientConnectorError
+from async_timeout import timeout
+from nextdns import ApiError, InvalidApiKeyError, NextDns
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_PROFILE_ID, CONF_PROFILE_NAME, DOMAIN
+
+
+class NextDnsFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for NextDNS."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.nextdns: NextDns
+        self.api_key: str
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+
+        websession = async_get_clientsession(self.hass)
+
+        if user_input is not None:
+            self.api_key = user_input[CONF_API_KEY]
+            try:
+                async with timeout(10):
+                    self.nextdns = await NextDns.create(
+                        websession, user_input[CONF_API_KEY]
+                    )
+            except InvalidApiKeyError:
+                errors["base"] = "invalid_api_key"
+            except (ApiError, ClientConnectorError, asyncio.TimeoutError):
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
```

Comment:
Omit this and only catch specific known exceptions

Reply:
What if there is an exception that we can't predict right now? I believe this is a good protection against unknown errors.

Reply:
It's against the design standards for the code base to catch raw Exception, isn't it? The problem is that it mixes programming errors with other types of errors.

Reply:
IMO sometimes we should skip standards when the situation requires it. Exactly 130 other integrations catches a general exception in config flow.
Mayby we should second opinion here 😀

Reply:
I think for a config flow it makes sense to catch broad exception to provide feedback to the user and not simply break the flow.

Reply:
I'm not seeing anything here that requires this (it seems arbitrary for this call.. any other line of could could also have a bug and throw an unexpected exception too) so happy to have someone else review and approve. Looks great otherwise.

Reply:
I wasn't aware this is part of the scaffold script template, which is helpful context.

https://github.com/home-assistant/core/blob/dev/script/scaffold/templates/config_flow/integration/config_flow.py#L92